samples/java/hibernate/src/main/java/com/google/cloud/postgres/HibernateSampleTest.java

```diff
@@ -0,0 +1,260 @@
+
+  private HibernateConfiguration hibernateConfiguration;
+
+  private List<String> singersId = new ArrayList<>();
+  private List<String> albumsId = new ArrayList<>();
+  private List<Long> tracksId = new ArrayList<>();
+  private List<String> venuesId = new ArrayList<>();
+  private List<String> concertsId = new ArrayList<>();
+
+  public HibernateSampleTest(
+      HibernateConfiguration hibernateConfiguration) {
+    this.hibernateConfiguration = hibernateConfiguration;
+  }
+
+  public void testJPACriteria() {
+    Session s = hibernateConfiguration.openSession();
+    CriteriaBuilder cb = s.getCriteriaBuilder();
+    CriteriaQuery<Singers> singersCriteriaQuery = cb.createQuery(Singers.class);
+    Root<Singers> singersRoot = singersCriteriaQuery.from(Singers.class);
+    singersCriteriaQuery
+        .select(singersRoot)
+        .where(
+            cb.and(cb.equal(singersRoot.get("firstName"), "David"),
+            cb.equal(singersRoot.get("lastName"), "Lee")));
+
+    Query<Singers> singersQuery = s.createQuery(singersCriteriaQuery);
+    List<Singers> singers = singersQuery.getResultList();
+
+    logger.log(Level.INFO, "Listed singer: {0}", singers.size());
+    s.close();
+  }
+
+  public void testHqlUpdate() {
+    Session s = hibernateConfiguration.openSession();
+
+    Singers singers = Utils.createSingers();
+    singers.setLastName("Cord");
+    s.getTransaction().begin();
+    s.saveOrUpdate(singers);
+    s.getTransaction().commit();
+
+    s.getTransaction().begin();
+    Query query = s.createQuery(
+        "update Singers set active=:active "
+            + "where lastName=:lastName and firstName=:firstName");
+    query.setParameter("active", false);
+    query.setParameter("lastName", "Cord");
+    query.setParameter("firstName", "David");
+    query.executeUpdate();
+    s.getTransaction().commit();
+
+    logger.log(Level.INFO, "Updated singer: {0}", s.get(Singers.class, singers.getId()));
+    s.close();
+  }
+
+  public void testHqlList() {
+    Session s = hibernateConfiguration.openSession();
+
+    Query query = s.createQuery("from Singers");
+    List<Singers> list = query.list();
+    logger.log(Level.INFO, "Singers list size: {0}", list.size());
+
+    query = s.createQuery("from Singers order by fullName");
+    query.setFirstResult(2);
+    list = query.list();
+    logger.log(Level.INFO, "Singers list size with first result: {0}", list.size());
+
+    /* Current Limit is not supported. */
+    // query = s.createQuery("from Singers");
+    // query.setMaxResults(2);
+    // list = query.list();
+    // logger.log(Level.INFO, "Singers list size with first result: {0}", list.size());
+
+    query = s.createQuery("select  sum(sampleRate) from Tracks");
+    list = query.list();
+    logger.log(Level.INFO, "Sample rate sum: {0}", list);
+    s.close();
+  }
+
+  public void testOneToManyData() {
+    Session s = hibernateConfiguration.openSession();
```

Comment:
nit: Is `Session` an instance of `AutoClosable`? If so, could we use `try-with-resources` blocks for these tests to adhere to Java best-practices.

Reply:
Done